modules/RTC/TPCUtils.js

```diff
@@ -88,40 +87,51 @@ export class TPCUtils {
      */
     _calculateActiveEncodingParams(localVideoTrack, codec, newHeight) {
         const codecBitrates = this.codecSettings[codec].maxBitratesVideo;
-        const height = localVideoTrack.getHeight();
-        const desktopShareBitrate = this.pc.options?.videoQuality?.desktopbitrate || DESKTOP_SHARE_RATE;
+        const trackHeight = localVideoTrack.getHeight();
+        const effectiveNewHeight = newHeight > trackHeight ? trackHeight : newHeight;
+        const desktopShareBitrate = this.pc.options?.videoQuality?.desktopbitrate || codecBitrates.ssHigh;
         const isScreenshare = localVideoTrack.getVideoType() === VideoType.DESKTOP;
         let scalabilityMode = this.codecSettings[codec].useKSVC
             ? VideoEncoderScalabilityMode.L3T3_KEY : VideoEncoderScalabilityMode.L3T3;
-        let maxBitrate = codecBitrates.high;
+        const qualityLevel = VIDEO_QUALITY_LEVELS.find(level => level.height === effectiveNewHeight);
+        let maxBitrate;
 
         if (this._isScreenshareBitrateCapped(localVideoTrack)) {
             scalabilityMode = VideoEncoderScalabilityMode.L1T3;
             maxBitrate = desktopShareBitrate;
         } else if (localVideoTrack.getVideoType() === VideoType.DESKTOP) {
             maxBitrate = codecBitrates.ssHigh;
+        } else if (qualityLevel) {
```

Comment:
If there is no defined qualityLevel we'll be setting `maxBitrate` to `undefined`. Is that ok?

When can that happen? Should we at least add a log line there?

Rounding to the nearest quality level sounds like the reasonable thing to do here IMHO, if we have no exact match.

480 is notably missing from the list of levels, for example.

Reply:
It wouldn't be catastrophic if maxBitrate is undefined, the browser would limit the bitrate to its defaults but you are right, rounding might be better especially for mobile. Will add 480 as well. Can you think of other resolutions that are applicable to mobile?

Reply:
I don't think I have seen any other weird ones. I'd add some log if we round to the nearest, after trying an exact match. This way we'll eventually see it while examining logs.